addOns/pscanrulesBeta/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/PiiScanner.java

```diff
@@ -95,30 +95,22 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                 Matcher matcher = cc.matcher(candidate.getCandidate());
                 while (matcher.find()) {
                     String evidence = matcher.group();
-                    if (validateLuhnChecksum(evidence) && !isSci(candidate.getContainingString())) {
+                    if (PiiUtils.isValidLuhn(evidence) && !isSci(candidate.getContainingString())) {
                         raiseAlert(msg, id, evidence, cc.name);
                     }
                 }
             }
         }
     }
 
-    private static boolean validateLuhnChecksum(String evidence) {
-        int sum = 0;
-        int parity = evidence.length() % 2;
-        for (int index = 0; index < evidence.length(); index++) {
-            int digit = Integer.parseInt(evidence.substring(index, index + 1));
-            if ((index % 2) == parity) {
-                digit *= 2;
-                if (digit > 9) {
-                    digit -= 9;
-                }
-            }
-            sum += digit;
-        }
-        return (sum % 10) == 0;
-    }
-
+    /**
+     * Checks whether a particular {@code String} input appears to be a valid number in scientific
+     * (exponent) notation. Ex: 2.14111111111111111e-2, 8.46786664623715E-47, 3.14111111111117293e5
+     *
+     * @param containingString the value to be checked.
+     * @return {@code true} if the value successfully parses as a {@code Float}, {@code false}
+     *     otherwise.
+     */
```

Comment:
I figured this javadoc may as well be kept in-case we come up with a reason to to split this out in the future.